subsys/bootloader/provision/provision_flash.c

```diff
@@ -11,17 +11,34 @@
 #include <nrf.h>
 #include <assert.h>
 #include <pm_config.h>
+#include <nrfx_nvmc.h>
+
 
 typedef struct {
 	u32_t s0_address;
 	u32_t s1_address;
 	u32_t num_public_keys;
-	u32_t pkd[1];
+	struct {
+		u32_t valid;
+		u32_t hash[CONFIG_SB_PUBLIC_KEY_HASH_LEN / 4];
```

Comment:
nit: `u8_t hash[CONFIG_SB_PUBLIC_KEY_HASH_LEN]`

Reply:
In that case I think we should change the signature of `public_key_data_read` to take u8_t as well.

    int public_key_data_read(u32_t key_idx, u8_t *p_buf, size_t buf_size)

Reply:
Added